packages/dataviews/src/components/validation/utils.ts

```diff
@@ -0,0 +1,40 @@
+import { ValidationContext } from '../../types';
+import { isRequiredRule, isRulesObject } from './guards';
+
+import { NormalizedRule, Rules } from './types';
+
+export const normalizeRules = < Item >(
+	isValidObject:
+		| ( ( item: Item, context?: ValidationContext ) => boolean | string )
+		| Rules< Item >
+		| undefined,
+	fieldId: string
+): NormalizedRule< Item > | undefined => {
+	if ( isValidObject && isRulesObject( isValidObject ) ) {
+		return generateCallback( isValidObject, fieldId );
+	}
+
+	if ( typeof isValidObject === 'function' ) {
+		return isValidObject;
+	}
+
+	return undefined;
+};
+
+const generateCallback = < Item extends unknown >(
+	rule: Rules< Item >,
+	fieldId: string
+) => {
+	if ( isRequiredRule( rule ) ) {
+		return ( item: any, context?: ValidationContext ) => {
+			const value = item[ fieldId ];
+			return value === undefined || value === ''
```

Comment:
How does this connect with the [field types](https://github.com/Automattic/wp-calypso/tree/trunk/packages/dataviews/src/field-types) we have? Does it need to? I'm not sure this function should be implemented at field type level, but it's something we may want to consider. It'd be more important when other type-bound rules are introduced.

For example, if a field is of type `email` it doesn't make sense to have the following (and we should clean the validation object before processing the field):

```js
{
  isValid: {
    min: 2,
    max: 4
  }
}
```

Reply:
Ideally, the type becomes mandatory, so we can do more things for the field author. However, field authors should be able to override mostly anything.